src/autotester/server/utils/tests/file_management_test.py

```diff
@@ -0,0 +1,424 @@
+                with tempfile.NamedTemporaryFile(dir=sub_dir2) as file:
+                    with patch(
+                        "autotester.server.utils.redis_management.test_script_directory",
+                        return_value=tmp_dir,
+                    ):
+                        yield tmp_dir, sub_dir1, sub_dir2, file.name
+
+
+def list_of_fd(file_or_dir):
+    """
+    Gets a list of files and directories and returns two separate list for files and directories
+    """
+    dir = []
+    files = []
+    for i in file_or_dir:
+        dir.append(i[1]) if i[0] == "d" else files.append(i[1])
+    return dir, files
+
+
+def test_clean_dir_name():
+    """
+    Checks whether '/' is replaced by '_' for a given path
+    """
+    a = "markus/address"
+    b = a.replace("/", "_")
+    assert clean_dir_name(a) == b
+
+
+def test_random_tmpfile_name():
+    """
+    Checks the temporary file name is random
+    """
+    tmp_file_1 = random_tmpfile_name()
+    tmp_file_2 = random_tmpfile_name()
+    assert not tmp_file_1 == tmp_file_2
+
+
+class TestRecursiveIglob:
+    """
+    Checks that all the files and directories of a given path are listed
+    """
+
+    def test_empty_dir(self, empty_dir):
+        """
+        When the Directory is empty
+        """
+        file_or_dir = list(recursive_iglob(empty_dir))
+        dir, files = list_of_fd(file_or_dir)
+        assert not dir and not files
+
+    def test_dir_has_onefile(self, dir_has_onefile):
+        """
+        When the Directory has only one file
+        """
+        root_dir, file = dir_has_onefile
+        file_or_dir = list(recursive_iglob(root_dir))
+        dir, files = list_of_fd(file_or_dir)
+        assert not dir
+        assert len(files) == 1 and file in files
+        assert os.path.exists(files[0])
+
+    def test_dir_has_onedir(self, dir_has_onedir):
+        """
+        When the Directory has only one sub directory
+        """
+        root_dir, sub_dir = dir_has_onedir
+        file_or_dir = list(recursive_iglob(root_dir))
+        dir, files = list_of_fd(file_or_dir)
+        assert not files
+        assert len(dir) == 1 and sub_dir in dir
+        assert os.path.exists(dir[0])
+
+    def test_dir_has_nestedfd(self, nested_fd):
+        """
+        When the files are nested in subdirectories more than 2 directories deep
+        """
+        root_dir, sub_dir1, sub_dir2, file = nested_fd
+        file_or_dir = list(recursive_iglob(root_dir))
+        dir, files = list_of_fd(file_or_dir)
+        assert all(os.path.exists(d) for d in dir)
+        assert all(os.path.exists(f) for f in files)
+        assert sub_dir1 in dir and sub_dir2 in dir
+        assert file in files
+
+
+class TestCopyTree:
+    """
+    Checks that all the contents are copied from source to destination
+    """
+
+    def test_empty_dir(self, empty_dir, dir_has_onefile):
+        """
+        When the Source Directory is empty
```

Comment:
Do not capitalize "Source Directory". (And similarly below.)

Also your description should include something about the destination directory.